Core/include/Acts/Utilities/RangeXD.hpp

```diff
@@ -0,0 +1,228 @@
+// This file is part of the Acts project.
+//
+// Copyright (C) 2021 CERN for the benefit of the Acts project
+//
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+#pragma once
+
+#include "Acts/Utilities/Range1D.hpp"
+
+#include <algorithm>
+#include <limits>
+
+namespace Acts {
+/// @brief An orthogonal range in an arbitrary number of dimensions
+///
+/// By combining a number one-dimensional ranges we can (under the assumption
+/// that our axes are orthogonal) construct an orthogonal range of values. In
+/// other words, a hyperrectangular volume in space.
+///
+/// @tparam Dims The number of dimensions in our range
+/// @tparam Type The scalar type of our ranges
+/// @tparam Vector The vector type used to define coordinates
+template <std::size_t Dims, typename Type,
+          template <typename, std::size_t> typename Vector = std::array>
+class RangeXD {
+ public:
+  /// @brief The type used to describe coordinates in our range
+  using coordinate_t = Vector<Type, Dims>;
+
+  /// @brief Determine whether this range is degenerate
+  ///
+  /// A degenerate multi-dimensional range has no volume and cannot contain any
+  /// values. This is the case if any of its dimensions are degenerate.
+  ///
+  /// @return true The range is degenerate
+  /// @return false The range is not degenerate
+  bool degenerate(void) const {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      if (m_dims[i].degenerate()) {
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /// @brief Determine whether the range contains a certain point
+  ///
+  /// This is true if and only if the range contains the point in all of its
+  /// dimensions.
+  ///
+  /// @param v The coordinate to check for membership in the range
+  ///
+  /// @return true The coordinate is inside the range
+  /// @return false The coordinate is outside the range
+  bool contains(const coordinate_t& v) const {
+    for (std::size_t i = 0; i < Dims; ++i) {
+      if (!m_dims[i].contains(v[i])) {
+        return false;
+      }
+    }
+
+    return true;
+  }
+
+  /// @brief Access one of the dimensional ranges of the volume
+  ///
+  /// @param i The index of the dimension to access
+  /// @return A reference to the dimension contained in this range
+  Range1D<Type>& operator[](const std::size_t& i) {
+    return m_dims[static_cast<std::size_t>(i)];
```

Comment:
Is that static_cast here necessary?
Looks like you have enforced the typ already.

Reply:
Well spotted, this is a vestige of some older design that can be removed.